libfwupd/fwupd-common.c

```diff
@@ -313,12 +313,15 @@ fwupd_build_user_agent (const gchar *package_name, const gchar *package_version)
 gchar *
 fwupd_build_machine_id (const gchar *salt, GError **error)
 {
+	const gchar *fn = "/etc/machine-id";
 	g_autofree gchar *buf = NULL;
 	g_autoptr(GChecksum) csum = NULL;
 	gsize sz = 0;
 
-	/* this has to exist */
-	if (!g_file_get_contents ("/etc/machine-id", &buf, &sz, error))
+	/* one of these has to exist */
+	if (!g_file_test (fn, G_FILE_TEST_EXISTS))
+		fn = "/var/lib/dbus/machine-id";
```

Comment:
Just a nitpick: Actually, if we want to make sure we are being extra correct I think we should replace `/var` with the `localstatedir` configured in meson. But this is such a niche I am not sure we care.

Reply:
I mean, dbus allows us to configure that. So I am thinking maybe we might want to do that too. But as I said, it's still a niche.

https://gitlab.freedesktop.org/dbus/dbus/blob/master/CMakeLists.txt#L72

Reply:
Please nitpick, it's positively encouraged! I'll fix up the patch.

Reply:
Okay, thanks :)

Reply:
I've added the prefixed and unprefixed versions too, as I think we should fall back to `/etc` if something like `/home/hughsie/.root/etc/machine-id` doesn't exist

Reply:
I am not sure if that is correct as `machine-id` should be a machine specific property and so I don't think users should be able to overwrite it. To me it's not making much sense but I might be wrong.

Reply:
but `localstatedir` seems to be affected when using `meson --prefix`

Reply:
Yes, but that for the d-bus machine ID. I think maybe we should keep the prefixed and unprefixed paths for `.../lib/dbus/machine-id` but it doesn't make sense for `/etc/machine-id`.

In https://www.freedesktop.org/software/systemd/man/machine-id.html they make a distinction from machine id and dbus machine id. 

> This machine ID adheres to the same format and logic as the D-Bus machine ID.

In https://dbus.freedesktop.org/doc/dbus-uuidgen.1.html they say the dbus id could change on reboot, which I don't think it's valid for the normal id.

> This will ensure that /var/lib/dbus/machine-id exists and has the uuid in it. It won't overwrite an existing uuid, since this id should remain fixed for a single machine until the next reboot at least.

Reply:
I am defenitely not an expert on this so correct me if I'm wrong. I haven't been able to find much information about this :confused:.